core/rawdb/accessors_l1_message_test.go

```diff
@@ -151,9 +153,83 @@ func TestIterationStopsAtMaxQueueIndex(t *testing.T) {
 	WriteHighestSyncedQueueIndex(db, 102)
 
 	// iteration should terminate at 102 and not read 103
-	got := ReadL1MessagesFrom(db, 100, 10)
+	got := ReadL1MessagesV1From(db, 100, 10)
 
 	if len(got) != 3 {
 		t.Fatal("Invalid length", "expected", 3, "got", len(got))
 	}
 }
+
+func TestIterateL1MessagesV1From(t *testing.T) {
+	msgs := []types.L1MessageTx{
+		newL1MessageTx(100),
+		newL1MessageTx(101),
+		newL1MessageTx(102),
+		newL1MessageTx(103),
+		newL1MessageTx(104),
+	}
+
+	db := NewMemoryDatabase()
+	WriteL1Messages(db, msgs)
+	WriteL1MessageV2StartIndex(db, 103)
+
+	it := IterateL1MessagesV1From(db, 100)
+	defer it.Release()
+
+	for _, msg := range msgs {
+		if msg.QueueIndex < 103 {
+			require.True(t, it.Next(), "Iterator terminated early")
+			require.Equal(t, msg.QueueIndex, it.L1Message().QueueIndex, "Invalid result")
+		} else {
+			require.Falsef(t, it.Next(), "Iterator did not terminate, queueIndex %d", msg.QueueIndex)
```

Comment:
A “knowledge sharing” related question about the iteration. do you know the collision mechanism (of different prefixes) of geth's state db?

Reply:
don't know. my goal with this test was to check whether the V1 iterator does not yield any message anymore after `V2StartIndex`

Reply:
It seems that collision is not an issue in this case. Because previously there was an issue that when iterating L1 messages with prefix `l1MessageLegacyPrefix = []byte("l1")`, it will go through some transactions stored in `txLookupPrefix = []byte("l")` because there are transactions which prefix "0x1", thus collide with each other.

While for:
```
l1MessagePrefix                   = []byte("L1") // l1MessagePrefix + queueIndex (uint64 big endian) -> L1MessageTx
l1MessageV2StartIndexKey          = []byte("L1MessageV2StartIndex")
l1MessageV2FirstL1BlockNumberKey  = []byte("L1MessageV2FirstL1BlockNumber")
```
It seems that it's impossible to collide with each other.